scripts/bash/run_enceladus.sh

```diff
@@ -289,7 +294,7 @@ fi
 
 SPARK_CONF="--conf spark.logConf=true"
 
-JVM_CONF="spark.driver.extraJavaOptions=-Dmenas.rest.uri=$MENAS_URI -Dstandardized.hdfs.path=$STD_HDFS_PATH \
+JVM_CONF="spark.driver.extraJavaOptions=-Dstandardized.hdfs.path=$STD_HDFS_PATH \
```

Comment:
Why was the rest URI removed?

Reply:
It is now placed in `application.conf` in the same directory as scripts. I'm gonna add the mentioning of in it to the `.template`.

Reply:
Just noticed, it is already there. Added a comment to that line.

Reply:
Are we now expecting users to provide their own app.conf? Or is it ok to use APPLICATION_PROPERTIES_CLIENT as well?

Reply:
Yes, we are asking users to provide their own `application.conf` and put all options there. 

Later we can remove our `application.template` logic and use `reference.conf` as you suggested.

Reply:
But it is not mandatory. Just easier. The older helper scripts will still work.

Instead of passing many options as `-D...`  to command line we just specify the location of the environment-specific `application.properties`. No credentials will now be passed to command line.

Reply:
Just a quick question. Did we try playing with `-Dconfig.resource=path/to/application.conf` option?

Reply:
Exactly. With the difference that we will be using `-Dconfig.file=path/to/application.conf`.

We need to distinguish between client and cluster mode. In client mode, the absolute path is used. In cluster mode, the config is sent via `--files ...` and read from the current directory.

Reply:
ok, I might be slow on the uptake here, so just to sum up. Using `config.file` in cluster mode does not propagate the config to the cluster. It need HDFS location.

Reply:
Not exactly. Let me explain.
1. `-Dconfig.file=/path/file` is a TypeSafe Config option saying where to look for the config.
2. When running in cluster mode the driver is started on a random cluster node, not on the edge node where `spark-submit` is called.
3. In order to put the configs to the node where the driver will be running `--files` option is used in `spark-submit`.
4. Files sent that way will appear on the driver's Yarn container in the current path. This includes `application.conf` sent this way.

Reply:
> Are we now expecting users to provide their own app.conf? Or is it ok to use APPLICATION_PROPERTIES_CLIENT as well?

I think I've misunderstood the question. Users will use the command line as always. But we will have per-environment `appllication.config` placed in the directory with the helper scripts.

Reply:
Great explanation. Thank you. Will not click `resolve conversation` as somebody might be interested as I was, but please treat it as resolved.

Reply:
I removed this since I removed it from our environments. And I want these scripts here to reflect as much as possible to what was actually deployed.

Reply:
> No sure why to remove the `application.conf.template` (?). I think it is a good example how to set the configuration values (and what are the configurations anyway).

reference.conf is a "template" when using lightbends typesafe config. If we move forward to using that we should follow their standards.

Reply:
I'd like to point out that there is a noteworthy downside to using `reference.conf` - that is, it makes it harder to make sure that you have overridden a default property from `reference.conf` in `application.conf` (e.g. mistyping a key will result in the default being used without warning). By using "only" `application.conf`, the config reading would fail fast instead.